internal/ethereum/blocklistener.go

```diff
@@ -470,23 +518,28 @@ func (bl *blockListener) dispatchToConsumers(consumers []*blockUpdateConsumer, u
 	}
 }
 
-func (bl *blockListener) checkStartedLocked() {
+func (bl *blockListener) checkStartedLocked(ctx context.Context) {
 	if bl.listenLoopDone == nil {
 		bl.listenLoopDone = make(chan struct{})
 		go bl.listenLoop()
 	}
+
+	bl.blockTillBlockFilterEstablished(ctx)
 }
 
 func (bl *blockListener) addConsumer(c *blockUpdateConsumer) {
 	bl.mux.Lock()
-	defer bl.mux.Unlock()
-	bl.checkStartedLocked()
 	bl.consumers[*c.id] = c
+	bl.mux.Unlock()
+	bl.checkStartedLocked(context.Background())
```

Comment:
We would propagate the correct context here

Reply:
and we need to keep it in the lock because we check the state of the block listener in checkStartedLocked with `bl.listenLoopDone == nil ` There could be a race condition where two loops get started if not

Reply:
@EnriqueL8, added a patch to lock till the listenLoopDone is set